orttraining/orttraining/python/orttraining_pybind_state.cc

```diff
@@ -160,6 +145,34 @@ TrainingConfigurationResult ConfigureSessionForTraining(
   return python_config_result;
 }
 
+#if defined(USE_NCCL)
+void CopyMPIContextToTrainingParameters(const onnxruntime::training::MPIContext& mpi_context,
+  TrainingParameters& parameters) {
+    std::cout << "mpi_context.world_rank: " << mpi_context.world_rank << std::endl;
```

Comment:
consider using "\n" instead of std::endl unless the flushing is really required.
is it possible to use ORT logging here?

Reply:
Good point! MPI is shutdown when a session is deleted. This is incorrect because there could be another training session. MPI should be shutdown only when the program is unloaded.

Reply:
updated with MPIContext as a singleton